v1.8: Distribution shift (#2814)
meilisearch/documentation: learn/experimental/vector_search.mdx

]'
```

### Distribution shift

> Hybrid search results do not always use the full vector space uniformly, which may impact relevancy. Use `distributionShift` when configuring an embedder to correct result distribution with an affine transformation:

Reviewer: Can we add [more context](https://meilisearch.notion.site/v1-8-AI-search-API-usage-135552d6e85a4a52bc7109be82aeca42#d652adc0890445658aaf36352dbc8802) to what those values mean, please?

> Semantic search results can sometimes have very close ranking scores. This may result in hybrid searches returning fewer keyword search results than expected. Use `distribution` when configuring an embedder to correct result distribution with an affine transformation:

Reviewer 2: From what I've been told, it can also lead to fewer semantic search results.

Reviewer 3: ```suggestion
For mathematical reasons, the `_rankingScore` of semantic search results tend to be closely grouped around an average value that depends on the embedder and model used.

This may result in relevant semantic hits being underrepresented and irrelevant semantic hits being overrepresented compared with keyword search hits.

For models that Meilisearch is aware of, that is, all OpenAI models and the Hugging Face model uses by default, Meilisearch already corrects the `_rankingScore` of semantic hits.

For other models, use `distribution` when configuring an embedder to correct the returned `_rankingScore`s of the semantic hits with an affine transformation.

To correctly use `distribution`:

1. First configure your embedder *without* the `distribution` setting
2. index your documents
3. Make several semantic search requests with the `showRankingScore` query parameter set to `true`.
4. Observe the results and manually decide if they are very relevant to your query, somewhat relevant to your query, or not relevant to your query.
5. Check the `_rankingScore`s of the results.
    1. the `_rankingScore`s of the "somewhat relevant" hits should be used as the `mean` of the `distribution` setting.
    2. the average difference between the `_rankingScore` of "very relevant" hits and "somewhat relevant" hits on one hand, and the `_rankingScore` of "somewhat relevant" hits and "not relevant" hists on the other hand should be used as the `sigma` of the `distribution` setting. 
```

```sh
curl \
  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
  -H 'Content-Type: application/json' \
  --data-binary '{
    "embedders": {
      "default": {
        "source":  "huggingFace",
        "model": "MODEL_NAME",
        "distribution": {
          "mean": 0.7,
          "sigma": 0.3
        }
      }
    }
  }'
```

> 

Author: ```suggestion

Configuring `distribution` requires a certain amount of trial and error, in which you must perform semantic searches and monitor the results. Based on their `rankingScore`s and relevancy, add the observed `mean` and `sigma` values for that index.

```

`distribution` is an optional field compatible with all embedder sources. It must be an object with two fields:

> - `mean`: a number between `0` and `1` indicating the new semantic score mean  

Reviewer 3: ```suggestion
- `mean`: a number between `0` and `1` indicating the semantic score of "somewhat relevant" hits before using the `distribution` setting.
```

> - `mean`: a number between `0` and `1` indicating the semantic score of "somewhat relevant" hits before using the `distribution` setting.

Author: ```suggestion
- `mean`: a number between `0` and `1` indicating the semantic score of "somewhat relevant" hits before using the `distribution` setting
```

> - `sigma`: a number between `0` and `1` indicating how much semantic score values may deviate from the mean. Use values smaller than 0.4 for larger variance, and larger than 0.4 to keep results close to the mean

Reviewer 3: ```suggestion
- `sigma`: a number between `0` and `1` indicating the average absolute difference in `_rankingScore`s between "very relevant" hits and "somewhat relevant" hits on one hand, and "somewhat relevant" hits and "irrelevant hits" on the other hand.
```
